flink-ai-flow/lib/airflow/airflow/api_connexion/openapi/v1.yaml

```diff
@@ -942,6 +942,56 @@ paths:
         '404':
           $ref: '#/components/responses/NotFound'
 
+  /dags/{dag_id}/dagRuns/{dag_run_id}/taskExecutions:
+    parameters:
+      - $ref: '#/components/parameters/DAGID'
+      - $ref: '#/components/parameters/DAGRunID'
+    get:
+      summary: List task executions
+      description: >
+        This endpoint allows specifying `~` as the dag_id, dag_run_id to retrieve DAG runs for all DAGs
+        and DAG runs.
+      x-openapi-router-controller: airflow.api_connexion.endpoints.task_execution_endpoint
+      operationId: list_task_executions
+      tags: [ TaskExecution ]
+      responses:
+        '200':
+          description: Success.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/TaskExecutionCollection'
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+
+  /dags/{dag_id}/dagRuns/{dag_run_id}/taskExecutions/{task_id}:
```

Comment:
IMO, the Restful API could be combined with `/dags/{dag_id}/dagRuns/{dag_run_id}/taskExecutions`.